### PR TITLE
Improvement made : by automating search engine configuration in example search_in_search_engine.py

### PR DESCRIPTION
**Pull request by Abhishek0075**
**Features**
This is a change made for improving new user's experience. When the examples are run there can be some confusion.
So when I went through the examples\search_with_specific_engine.py there is a need for adding the api key and cse id explicitly and also need to uncomment and comment the code snippets according to our search engine preference
Through my updation i have made it easy by just using the config2.yaml configuration and doing the run
The user just need to run the program no need to bother about the code changes according to configurations. 

**Influence**
It can make the new users' experience great and ease the usage of examples and understanding the code

### CODE REVIEW

File: examples/debate.py [hunk 1]
```diff
@@ -34,7 +34,9 @@ class SpeakAloud(Action):
     name: str = "SpeakAloud"
 
     async def run(self, context: str, name: str, opponent_name: str):
-        prompt = self.PROMPT_TEMPLATE.format(context=context, name=name, opponent_name=opponent_name)
+        prompt = self.PROMPT_TEMPLATE.format(
```
Comment (iorisa) [inline examples/debate.py:37]: Run the commands below to fix coding standards：

```shell
pip install pre-commit
pre-commit install
pre-commit run --all-files  # make sure all files are clean
```

File: examples/search_with_specific_engine.py [hunk 2]
```diff
@@ -4,21 +4,37 @@
...
+
+    search = Config.default().search
+    kwargs = {"api_key": search.api_key, "cse_id": search.cse_id, "proxy": None}
+
+    if search.api_type == SearchEngineType.DIRECT_GOOGLE:
+        # Google API
+        await Searcher(
+            search_engine=SearchEngine(engine=SearchEngineType.DIRECT_GOOGLE, **kwargs)
+        ).run(question)
+    elif search.api_type == SearchEngineType.SERPER_GOOGLE:
```
Comment (iorisa) [inline examples/search_with_specific_engine.py:23]: You can replace these if-else statements with a dict to reduce complexity, like this:
```python
mappings = {
    SearchEngineType.DIRECT_GOOGLE: {
        "api_key": search.api_key, 
        "cse_id": search.cse_id, 
        "proxy": None, 
        "engine":SearchEngineType.DIRECT_GOOGLE}, 
     ...}
args = mappings.get(search.api_type)
search_engine=SearchEngine(**args)
```

Comment (Abhishek0075) [reply examples/search_with_specific_engine.py:23]: search = Config.default().search
kwargs = {"api_key": search.api_key, "cse_id": search.cse_id, "proxy": None}await Searcher(search_engine=SearchEngine(engine=search.api_type, **kwargs)).run(question)
 
    It only needs this right when you just made this suggestion I looked into this

Comment (iorisa) [reply examples/search_with_specific_engine.py:23]: `examples/search_with_specific_engine.py` after running pre-commit:
```python
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
"""
import asyncio

from metagpt.config2 import Config
from metagpt.roles import Searcher
from metagpt.tools.search_engine import SearchEngine, SearchEngineType


async def main():
    question = "What are the most interesting human facts?"

    search = Config.default().search
    kwargs = {"api_key": search.api_key, "cse_id": search.cse_id, "proxy": None}

    if search.api_type == SearchEngineType.DIRECT_GOOGLE:
        # Google API
        await Searcher(search_engine=SearchEngine(engine=SearchEngineType.DIRECT_GOOGLE, **kwargs)).run(question)
    elif search.api_type == SearchEngineType.SERPER_GOOGLE:
        # Serper API
        await Searcher(search_engine=SearchEngine(engine=SearchEngineType.SERPER_GOOGLE, **kwargs)).run(question)
    elif search.api_type == SearchEngineType.SERPAPI_GOOGLE:
        # SerpAPI
        await Searcher(search_engine=SearchEngine(engine=SearchEngineType.SERPAPI_GOOGLE, **kwargs)).run(question)
    else:
        # DDG API
        await Searcher(search_engine=SearchEngine(engine=SearchEngineType.DUCK_DUCK_GO, **kwargs)).run(question)


if __name__ == "__main__":
    asyncio.run(main())
```

Comment (iorisa) [reply examples/search_with_specific_engine.py:23]: This is the cyclomatic complexity after cascading if-else statements:
```shell
radon cc -a examples/search_with_specific_engine.py
examples/search_with_specific_engine.py
    F 12:0 main - A

1 blocks (classes, functions, methods) analyzed.
Average complexity: A (4.0)
```
In the case of this code, the cyclomatic complexity is acceptable.

Comment (Abhishek0075) [reply examples/search_with_specific_engine.py:23]: This is the latest complexity showing when i have made(latest commit on 1 the above mentioned change

radon cc -a search_with_specific_engine.py
search_with_specific_engine.py
    F 12:0 main - A

1 blocks (classes, functions, methods) analyzed.
Average complexity: A (1.0)

**Comment by iorisa** [comment]
LGTM

**Pull request merged**